### ISSUE TITLE
Dictionary plugin started failing due to certificate issues

### ISSUE THREAD

**Issue by NotWearingPants** [oliverschwendener/ueli#807]
This started to happen only a couple of days ago.
The dictionary worked fine, but now it shows the following for every query:

![image](https://user-images.githubusercontent.com/26556598/135551131-6b5ed6a1-1345-4289-b37d-fa5d97a9906f.png)

And in the debug log I'm getting `error: certificate has expired`.

I tried accessing the dictionary API myself with axios and it works fine.
Their certificate is from 2021-09-10, seems to be by LetsEncrypt which swaps certificates every 3 months.

This also happens with the weather plugin, but not the currency or translation plugins.

Is there some sort of certificate caching going on in Electron/Node/Axios?
Is this happening to you too?

Running "clear caches" didn't help, wiping ueli's entire data folder (%APPDATA%\ueli) didn't help either.

**Comment by fedegblack** [comment]
Yes, same issue on my end.

**Comment by dansroot** [comment]
The same seems to happen when using the wather plugin (wea?).

**Comment by Karan-Munjani** [comment]
@NotWearingPants How can I debug the app? I've never worked on electron but happy to help if I can solve the issue!

**Comment by tkohlmeier** [comment]
Seems to be the same issue as here: https://stackoverflow.com/questions/69403020/axios-certificate-has-expired-on-valid-certificate

Probably because Let's Encrypt switched to a new root certificate: https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/

I don't know which package or component actually causes this, but we'll probably have to wait for an update.

**Comment by tkohlmeier** [comment]
It's already [fixed in Electron](https://github.com/electron/electron/pull/31213). But only for the [supported versions (12-15)](https://www.electronjs.org/docs/tutorial/support#supported-versions). Ueli uses Electron 11, so we need to update to get the fix.

**Comment by NotWearingPants** [comment]
**TL;DR**: Let's Encrypt is a big certificate supplier, announced a significant change six months ago. OpenSSL prepared for it but boringssl were like: \*surprised pikachu face\*. Sadly Electron uses boringssl.

---

I've just spent an hour debugging it and found the core issue only to come back and see you've figured it out, whoops.

To summarize:
Let's Encrypt [announced in April 2021](https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/) that it's going to switch up some of its certificate chain on September 30. Ironically I heard about this in April, thought people were fixing it.

But only after it happened people started noticing, and an [issue](https://bugs.chromium.org/p/boringssl/issues/detail?id=439) was opened in boringssl, Google's SSL library. and it was [fixed](https://boringssl.googlesource.com/boringssl/+/8f5eb80b810ff63d14ad3535cb16f7cb8271a4f5) the next day.

Electron [noticed this earlier](https://github.com/electron/electron/issues/31212) and fixed it in [main](https://github.com/electron/electron/pull/31213) by patching boringssl instead of waiting for them.
They backported the fix to [12.2.1](https://github.com/electron/electron/pull/31214), [13.5.1](https://github.com/electron/electron/pull/31215), [14.1.0](https://github.com/electron/electron/pull/31216), [15.1.0](https://github.com/electron/electron/pull/31218), and [16 beta](https://github.com/electron/electron/pull/31217), but not 11 unfortunately as it is already unsupported.

[Electron versions](https://www.electronjs.org/docs/tutorial/electron-timelines) reach end-of-life 10 months after their initial release, and 11 was released in 2020-11-17. When 16 is released in 2021-11-16 Electron 12 will also stop getting updates.

**Comment by NotWearingPants** [comment]
> Update: Just tried updating to Electron 12. It works when BrowserWindows are created with webPreferences.contextIsolation: false. And the certificate error is gone then. But we probably should make some changes so that contextIsolation can remain enabled.

I think the best course of action is to update ASAP to fix the dictionary & weather issues, and worry about `contextIsolation` later.

I've gone over all of Electron's breaking changes up to version 15, and the only ones affecting ueli are `contextIsolation` which defaults to `true` since v12, and the `remote` module which is deprecated since v12 and removed in v14.

ueli specifies `enableRemoteModule: true` but never uses the `remote` module anywhere, yet I've tried removing this option and it breaks the renderers entirely. Not sure why, need to debug this later.

So I recommend upgrading to Electron 13 since 14 doesn't work, and add `contextIsolation: false` and worry about security later since this doesn't change anything currently.

Regarding security ueli also has `nodeIntegration` enabled, which is considered unsafe. Although these safety issues aren't really a concern for ueli IMO, it never executes any code that comes from outside of the app.

**Comment by oliverschwendener** [comment]
This should be fixed with the latest release.
